pkg/flow/database.go

```diff
@@ -517,15 +517,26 @@ func (internal *internal) getInstance(ctx context.Context, inc *ent.InstanceClie
 		return nil, err
 	}
 
-	query := inc.Query().Where(entinst.IDEQ(id)).WithNamespace().WithWorkflow()
+	query := inc.Query().Where(entinst.IDEQ(id)).WithNamespace().WithWorkflow(func(q *ent.WorkflowQuery) {
+		q.WithInode()
+	})
 	if load {
 		query = query.WithRuntime()
 	}
 	in, err := query.Only(ctx)
+	if err != nil {
+		return nil, err
+	}
 
 	d := new(instData)
 	d.in = in
 
+	// TODO: check if this is a good place to put this
```

Comment:
Did you mean to do this TODO in this pr, or is this TODO for the future?

Reply:
Happy to leave it here for now. I think the code would be more maintainable if I put this logic here, but also possibly less performant.